src/custom-element.js

```diff
@@ -1146,12 +1151,17 @@ function createBaseCustomElementClass(win) {
      *
      * Can only be called on a upgraded and built element.
      *
+     * @param {!AbortSignal} signal
      * @return {!Promise}
      * @package @final
      */
-    layoutCallback() {
+    layoutCallback(signal) {
       assertNotTemplate(this);
       devAssert(this.isBuilt(), 'Must be built to receive viewport events');
+      if (signal.aborted) {
+        throw cancellation();
```

Comment:
Can we instead to `Promise.reject(cancellation())`?

Reply:
Done.